### PR TITLE
Add filter option for data extent

### PR DESCRIPTION
**Pull request by wkramer**

### CODE REVIEW

File: src/Charts/chart.ts [hunk 1]
```diff
@@ -60,14 +61,19 @@ export enum CurveType {
...
 }
 
-export interface ChartOptions {
+export interface ChartOptionsForKeys {
   x?: ChartOptionItem
   x1?: ChartOptionItem
   y?: ChartOptionItem
   radial?: ChartOptionItem
   angular?: ChartOptionItem
-  color?: ColorOptionItem
```
Comment (ceesvoesenek) [inline src/Charts/chart.ts:69]: Why has this been deleted?

File: src/Charts/chart.ts [hunk 2]
```diff
@@ -176,7 +182,12 @@ export abstract class Chart {
     return this
   }
 
-  setOptions(options: ChartOptions) {
+  setOptions(options: ChartOptions) {    
+    for (const key of chartKeys) {
```
Comment (ceesvoesenek) [inline src/Charts/chart.ts:186]: Why is this necessary now?

File: src/Charts/chart.ts [hunk 3]
```diff
@@ -127,11 +133,11 @@ export abstract class Chart {
...
   get extent(): Record<string, number[] | Date[] | null[]> {
-    if (!this._extent) {
-      this._extent = []
-      for (const key in this.dataKeys) {
-        const path = this.dataKeys[key]
-        this._extent[path] = dataExtentFor(this._data, path)
+    if (!this._extent) this._extent = {}
+    for (const key in this.dataKeys) {
+      const path = this.dataKeys[key]
+      if (path in this._extent || this._extent[path] === undefined) {
```
Comment (ceesvoesenek) [inline src/Charts/chart.ts:139]: Is this not always true? If path is not in `this._extent`, `this._extent[path]` will always be `undefined`?

Comment (wkramer) [reply src/Charts/chart.ts:139]: The second gives an error if path is not a propery of this._extent

File: src/Charts/chart.ts (hunk 2 above)
Comment (wkramer) [reply src/Charts/chart.ts:186]: Because you can now pass `setOtions({ y :  { extentFilter: () => false })`. Which must invalidate the stored extent.

File: src/Charts/chart.ts (hunk 3 above)
Comment (ceesvoesenek) [reply src/Charts/chart.ts:139]: Is `this._extent` not a regular JavaScript object? Also, in that case it will always error if `path` is not in `this._extent`, since it is `||` and not `&&`?
